engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/Interpreter.scala

```diff
@@ -57,9 +57,9 @@ private class InterpreterInternal[F[_]](listeners: Seq[ProcessListener],
 
   private def interpretNode(node: Node, ctx: Context): F[List[InterpretationResult]] = {
     implicit val nodeImplicit: Node = node
-    listeners.foreach(_.nodeEntered(node.id, ctx, metaData))
     node match {
       case Source(_, next) =>
+        listeners.foreach(_.nodeEntered(node.id, ctx, metaData))
```

Comment:
Doesn't we count twice `nodeEntered` for sources? You have it here and in `interpretNext`

Reply:
No, in `interpretNext` I invoke this method for **next** node, not current one

For what I know, it is not possible to have another node before soruce in the graph - only in this case it would be counted twice.